.github/workflows/image_build_push.yml

```diff
@@ -0,0 +1,172 @@
+  DOCKER_PASSWORD: ${{secrets.DOCKER_PASSWORD}}
+
+jobs:
+  fetch_run_id:
+    runs-on: ubuntu-latest
+
+    outputs:
+      run_id: ${{ steps.get_run_id.outputs.run_id }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with: 
+          python-version: '3.x'
+
+      - name: Install Python dependencies
+        run: |
+         python -m pip install --upgrade pip
+         pip install requests
+
+      - name: Run Python script
+        id: run_script
+        run: |
+          echo "Fetching latest successful run ID from e-misison-server docker image workflow"
+          python .github/fetch_runID.py
+    
+      - name: Get Run ID
+        id: get_run_id
+        run: echo "run_id=${{ steps.run_script.outputs.run_id }}" >> "$GITHUB_OUTPUT"
+
+  fetch_tag:
+    needs: fetch_run_id
+    runs-on: ubuntu-latest
+
+    env:
+      RUN_ID: ${{ needs.fetch_run_id.outputs.run_id }}
+
+    outputs:
+      docker_image_tag: ${{ steps.get_docker_tag.outputs.docker_image_tag }}
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Use Run ID from previous fetch_run_id job
+      run: echo Run ID from previous job ${{ env.RUN_ID }}
+      
+    - name: Download artifact
+      uses: actions/download-artifact@v4
+      with:
+        name: docker-image-tag
+        # TODO: Create a token with basic repo permissions
+        github-token: ${{ secrets.GH_PAT_TAG }}
+        repository: e-mission/e-mission-server
+        run-id: ${{ env.RUN_ID }}
+
+    - name: Print artifact tag
+      id: get_docker_tag
+      run: |
+        cat tag_file.txt
+        docker_image_tag=$(cat tag_file.txt)
+        echo $docker_image_tag
+        echo "docker_image_tag=$(echo $docker_image_tag)" >> $GITHUB_OUTPUT
+
+  build:
+    needs: fetch_tag
+    runs-on: ubuntu-latest
+
+    env:
+      DOCKER_IMAGE_TAG_1: ${{ needs.fetch_tag.outputs.docker_image_tag }}
+      DOCKER_IMAGE_TAG_2: ${{ github.event.inputs.docker_image_tag }}
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Print input docker image tag
+      run: |
+        echo "Event name: ${{ github.event_name }}"
+        echo "Latest docker image tag (push): ${{ env.DOCKER_IMAGE_TAG_1 }}"
+        echo "Latest docker image tag (workflow_dispatch): ${{ env.DOCKER_IMAGE_TAG_2 }}"
+
+    - name: Update .env file 
+      run: |
+        if [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
+          echo "Workflow_dispatch: New server image built and pushed, Updating image tag in .env"
+          echo "SERVER_IMAGE_TAG=$DOCKER_IMAGE_TAG_2" > .env
+          echo "COGNITO_CLIENT_ID=''" > .env
+          echo "COGNITO_CLIENT_SECRET=''" > .env
+          echo "COGNITO_REDIRECT_URL=''" > .env
+          echo "COGNITO_TOKEN_ENDPOINT=''" > .env
+          echo "COGNITO_USER_POOL_ID=''" > .env
+          echo "COGNITO_REGION=''" > .env
+          echo "COGNITO_AUTH_URL=''" > .env
```

Comment:
I am confused. If the goal is to have all the environment variables stored in `.env`, then why are you overriding them here? And if the goal is not have them in here, then why do we need to fill them in?

Basically, if you are going to override the values with empty strings anyway, then it is not clear that you need the variables in the first place.

Reply:
@nataliejschultz Had added these to the .env file in [this commit](https://github.com/MukuFlash03/op-admin-dashboard/commit/a93760165caae8c36857b5d6fc4d3dc70a5ce8cb).

The reason for doing this was that GitHub Actions was overwriting the .env file as a part of the workflow. During the workflow only the latest server tag is written into the .env file.

Reply:
But I do see a problem with this:
- If we make Cognito variables as empty strings in workflow, original stored values will be lost.

Initially, when we built the logic for the CI/CD to update the server image tag, the .env file was meant to have just the server image tag (related commits: [1](https://github.com/MukuFlash03/op-admin-dashboard/commit/d98f75c8ba36a8b69c547f30e8840b6af773c7a5), [2](https://github.com/MukuFlash03/op-admin-dashboard/commit/d03ce01665757a97576920db600c89c18d0b3f51)).

Reply:
Added a separate template file in [this commit](https://github.com/e-mission/op-admin-dashboard/pull/113/commits/3f66ef344a70aebf11913b6202a007b051f226c1).
Keeping the .env file only for latest server image tag.

Reply:
@MukuFlash03 Are you sure that multiple `.env` files are supported in docker? How did you test this to make sure that if the values are set, they are visible in the codebase?

Reply:
I don't think they would be set directly.

I thought of this approach, I had mentioned this in the commit:
```
I am expecting the users would either set them directly in the system or use the docker compose prod yml file.
```
So, with this option, the cognito env file would literally just be a template.

---

The second option is that we will need to load the env variables from the specified file using the python library [python-dotenv](https://pypi.org/project/python-dotenv/). Will need to add this in requirements.txt.

So, the code in utils/cognito_utils.py (for instance) would look like:
```
from dotenv import load_dotenv

load_dotenv(.env.cognito)

def get_tokens(code):
    client_id = os.getenv("COGNITO_CLIENT_ID", '')
    client_secret = os.getenv("COGNITO_CLIENT_SECRET", '')
    redirect_uri = os.getenv("COGNITO_REDIRECT_URL", '')
    token_endpoint = os.getenv("COGNITO_TOKEN_ENDPOINT", '')

```

Reply:
`python-dotenv` is overkill
I guess it is fair to assume that the users will set the values in the environment, we do assume that for the server code. Once we figure out how to maintain the list of environment variables from the server, we should use the same approach here. Ideally this would be through a search so that we don't have to maintain a separate template file and ensure that it doesn't bitrot.

Reply:
Note also that we don't have to support a `.env` with a single tag. There are ways of editing files programmatically that don't involve `>>`